### PR TITLE
Don't rename 'unnamed' tabs upon deletion of other tabs

### PR DESCRIPTION
**Pull request by AdaShoelace**
This PR aims to solve issue #474.
I decided to put the logic in `Tab::new`. This approach felt the "cleanest" to me.
If this breaks any style guides or should be done differently for other reasons please let me know and I'll change it.

### CODE REVIEW

File: zellij-server/src/tab.rs [hunk 1]
```diff
@@ -259,7 +259,11 @@ impl Tab {
             index,
             position,
             panes,
-            name,
+            name: if name.is_empty() {
```
Comment (a-kenji) [inline zellij-server/src/tab.rs:262]: I would prefer the initialization here just like the `panes` are initialized.

`let name =` prior and then keep the name here.

**Review by a-kenji — CHANGES_REQUESTED** [review]
Thank you, this is already pretty great and
I think including it in `tab` does make a lot of sense!

There is still one thing that I would like to see a little bit different:

When opening a tab and not really caring about the name,
I think I know what you want to convey with the (unnamed), but I think it
does take up too much space. And should be fine with being omitted.

**Comment by AdaShoelace** [comment]
> Thank you, this is already pretty great and
> I think including it in `tab` does make a lot of sense!
> 
> There is still one thing that I would like to see a little bit different:
> 
> When opening a tab and not really caring about the name,
> I think I know what you want to convey with the `(unnamed)`, but I think it
> does take up too much space. And should be fine with being omitted.

"(unnamed)" has been removed.

**Review by a-kenji — APPROVED** [review]
Awesome, thank you!

**Comment by ammkrn** [comment]
@a-kenji 

It seems like some underlying map of tabs to numbers still exhibits the old behavior. If I have `tab1, tab2, tab3, tab4` and close`tab2`, the remaining tabs are `tab1, tab3, tab4`, but in tab mode, pressing the `2` key jumps to `tab3` and the `3` key jumps to `tab4`.

If other people are observing this I can take a look at it. Hints for where to look would be appreciated.

**Comment by a-kenji** [comment]
@ammkrn,
This is correct, but sadly still confusing behaviour.
The tabs are now named that way by default, but the keys
map to the position in the tab. I hope with #557 we can show 
the number of the tab and then the name of the tab with a 
running program/shell behind.

I believe we want (am happy about more input):
To go to a specific tab-name: #379. (An ongoing pr atm).
To go to a specific tab#. (The current 2 behaviour).

I imagined this here to be in scope of #557:
To show the number of the current tab.
To show the name of the current tab. (possibly with name of currently running process)

**Comment by ammkrn** [comment]
Thanks for clearing that up; I look forward to seeing how those solutions work out.

**Pull request merged**